core-lightning-squeaknode/docker-compose.yml

```diff
@@ -0,0 +1,51 @@
+version: "3.7"
+
+services:
+  app_proxy:
+    environment:
+      APP_HOST: $APP_CORE_LIGHTNING_SQUEAKNODE_IP
+      APP_PORT: $APP_CORE_LIGHTNING_SQUEAKNODE_PORT
+
+  web:
+    image: ghcr.io/squeaknode/squeaknode:v0.3.4@sha256:c5b8b4326b20e487ec0366a4e8af27f8aa894421d267a8e6d78916c8be765e7f
+    restart: on-failure
+    stop_grace_period: 1m
+    ports:
+      - "$APP_CORE_LIGHTNING_SQUEAKNODE_GRPC_PORT:$APP_CORE_LIGHTNING_SQUEAKNODE_GRPC_PORT"
+    volumes:
+      - ${APP_DATA_DIR}/sqk:/sqk
+      - ${APP_CORE_LIGHTNING_RPC_FILE_PATH}:/c-lightning/lightning-rpc
```

Comment:
This env var won't exist if the user hasn't updated to the newer version of Core Lightning where we've added this new env var to it's exports file.

This will cause issues because it'll resolve to `.` which will mean Squeaknode will essentially break out of the app sandbox and get access to a lot more stuff than it should be allowed to.

Also generally mounting files into Docker as volumes is not a good idea because if the file doesn't exist Docker will automatically create a directory in the same place. I'm not 100% sure but I'd guess the `lightning-rpc` file doesn't exist when Core Lightning starts and it would create it as part of it's startup process. If that's the case this is going to create a race condition on each Umbrel start where if Squeaknode starts before Core Lightning has already started and created it's `lightning-rpc` file Docker will create an empty directory in it's place and Core Lightning will probably have issues starting because of that.

I'm honestly not sure of the best way to resolve this. One solution to resolve the race condition would be to mount the entire Core Lightning dir in, but that's suboptimal since then Squeaknode gets access to Core Lightning private keys. We also don't fix the missing env var issue.

Is Squeaknode able to interact with Core Lightning's new gRPC port? Maybe that would be a cleaner way to interface. I believe It's the same protocol as the unix socket but just exposed over gRPC.

If Squeaknode could interface with Core Lightning over it's network interface then that would resolve both the race condition and the sandbox issue for old versions.

Reply:
@lukechilds  Another option is to use the `–rpc-file` option when you start core-lightning to specify a different location for the rpc file outside of that directory.

Reply:
@lukechilds I updated the PR so now instead of `APP_CORE_LIGHTNING_RPC_FILE_PATH` it exports `APP_CORE_LIGHTNING_RPC_FILE_DIR`.

I added the `--rpc-file` config to core-lightning, and created a new rpc directory to hold the rpc file with a .gitkeep.